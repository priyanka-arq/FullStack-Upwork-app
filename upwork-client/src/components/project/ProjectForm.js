import React, { Component } from "react";
import TagsInput from "react-tagsinput";
import "react-tagsinput/react-tagsinput.css";
import AuthService from "../../services/auth.service";

class ProjectForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      skills: this.props.project ? this.props.project.skills : [],
      name: this.props.project ? this.props.project.name : "",
      phone: this.props.project ? this.props.project.phone : "",
      email: this.props.project ? this.props.project.email : "",
      description: this.props.project ? this.props.project.description : "",
      paymentType: this.props.project ? this.props.project.paymentType : "",
      estimatedBudget: this.props.project
        ? this.props.project.estimatedBudget
        : 0,
      redirect: null,
      userReady: false,
      currentUser: {},
      tags: [],
    };
  }

  componentDidMount() {
    const currentUser = AuthService.getCurrentUser();
    if (!currentUser) this.setState({ redirect: "/login" });
    this.setState({ currentUser: currentUser, userReady: true });
  }

  handleSkills = (tags) => {
    this.setState({ skills: tags });
  };

  onSubmit = (event) => {
    event.preventDefault();

    this.props.onSubmit({
      skills: this.state.skills,
      name: this.state.name,
      description: this.state.description,
      paymentType: this.state.paymentType,
      phone: this.state.phone,
      email: this.state.email,
      estimatedBudget: this.state.estimatedBudget,
      user: this.state.currentUser ? this.state.currentUser.id : undefined,
    });
  };

  render() {
    return (
      <div>
        {this.state.userReady ? (
          <div className="form-group">
            <form onSubmit={this.onSubmit}>
              <h1>New Project</h1>
              <h3>Name for your project</h3>
              <input
                id="name-input"
                className="form-control"
                onChange={(e) => this.setState({ name: e.target.value })}
                value={this.state.name}
                placeholder="name"
                required
              />

              <h3>Tell us more about your project</h3>
              <textarea
                className="form-control"
                onChange={(e) => this.setState({ description: e.target.value })}
                value={this.state.description}
                type="text"
                placeholder="project description"
                required
              />

              <h3>Skills Required</h3>
              <TagsInput
                className="form-control"
                onChange={this.handleSkills}
                value={this.state.tags}
                type="text"
                placeholder="Skills "
                required
              />

              <h3>Payment Type</h3>
              <select
                className="form-control"
                onChange={(e) => this.setState({ paymentType: e.target.value })}
                value={this.state.paymentType}
                type="text"
                placeholder="payment"
                required
              >
                <option value="">Select paymentType</option>
                <option value="Fixed-Price">Fixed Price</option>
                <option value="Hourly-Rate">Hourly Rate</option>
              </select>

              <h3>Estimated Budget</h3>
              <input
                className="form-control"
                onChange={(e) =>
                  this.setState({ estimatedBudget: e.target.value })
                }
                value={this.state.estimatedBudget}
                type="number"
                placeholder="$"
                required
              />

              <h3>Phone Number</h3>
              <input
                className="form-control"
                onChange={(e) => this.setState({ phone: e.target.value })}
                value={this.state.phone}
                type="number"
                placeholder="Phone"
                required
              />

              <h3>Email</h3>
              <input
                className="form-control"
                onChange={(e) => this.setState({ email: e.target.value })}
                value={this.state.email}
                type="email"
                placeholder="email"
                required
              />

              <button className="button" onClick={this._handleCreate}>
                Post Project
              </button>
            </form>
          </div>
        ) : null}
      </div>
    );
  }
}
export default ProjectForm;
