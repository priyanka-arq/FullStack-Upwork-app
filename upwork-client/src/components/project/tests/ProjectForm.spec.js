import React from "react";
import ProjectForm from "../../project/ProjectForm";
import { shallow } from "enzyme";
import projects from "../../../fixtures/projects";

let wrapped, onSubmitSpy, userReady, currentUser;

beforeEach(() => {
  onSubmitSpy = jest.fn();
  currentUser = { id: 1 };
  userReady = true;
  wrapped = shallow(
    //testing connected version of Profile by wrapping it with <Root />

    <ProjectForm
      currentUser={currentUser}
      userReady={userReady}
      onSubmit={onSubmitSpy}
      project={projects[0]}
    />
  );
});

afterEach(() => {
  wrapped.unmount();
});

test("Should render project form correctly ", () => {
  expect(wrapped).toMatchSnapshot();
});

describe("render profileForm value", () => {
  it("should set title on input change", () => {
    const value = "portfoio";
    wrapped.find("input").at(0).simulate("change", {
      target: { value },
    });

    expect(wrapped.state("name")).toEqual("portfoio");
  });

  test("should call onSubmit prop for valid form submission", () => {
    wrapped.find("form").simulate("submit", {
      preventDefault: () => {},
    });
    expect(onSubmitSpy).toHaveBeenCalledTimes(1);
    expect(onSubmitSpy).toHaveBeenLastCalledWith({
      skills: projects[0].skills,
      name: projects[0].name,
      description: projects[0].description,
      paymentType: projects[0].paymentType,
      phone: projects[0].phone,
      email: projects[0].email,
      estimatedBudget: projects[0].estimatedBudget,
      // user: currentUser.id,
    });
  });
});
