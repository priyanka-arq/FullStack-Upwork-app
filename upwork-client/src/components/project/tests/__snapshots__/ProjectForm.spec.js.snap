// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render project form correctly  1`] = `
<div>
  <div
    className="form-group"
  >
    <form
      onSubmit={[Function]}
    >
      <h1>
        New Project
      </h1>
      <h3>
        Name for your project
      </h3>
      <input
        className="form-control"
        id="name-input"
        onChange={[Function]}
        placeholder="name"
        required={true}
        value="TEST1"
      />
      <h3>
        Tell us more about your project
      </h3>
      <textarea
        className="form-control"
        onChange={[Function]}
        placeholder="project description"
        required={true}
        type="text"
        value="TEST1"
      />
      <h3>
        Skills Required
      </h3>
      <TagsInput
        addKeys={
          Array [
            9,
            13,
          ]
        }
        addOnBlur={false}
        addOnPaste={false}
        className="form-control"
        disabled={false}
        focusedClassName="react-tagsinput--focused"
        inputProps={Object {}}
        maxTags={-1}
        onChange={[Function]}
        onlyUnique={false}
        pasteSplit={[Function]}
        placeholder="Skills "
        preventSubmit={true}
        removeKeys={
          Array [
            8,
          ]
        }
        renderInput={[Function]}
        renderLayout={[Function]}
        renderTag={[Function]}
        required={true}
        tagDisplayProp={null}
        tagProps={
          Object {
            "className": "react-tagsinput-tag",
            "classNameRemove": "react-tagsinput-remove",
          }
        }
        type="text"
        validationRegex={/\\.\\*/}
        value={Array []}
      />
      <h3>
        Payment Type
      </h3>
      <select
        className="form-control"
        onChange={[Function]}
        placeholder="payment"
        required={true}
        type="text"
        value="Fixed-Price"
      >
        <option
          value=""
        >
          Select paymentType
        </option>
        <option
          value="Fixed-Price"
        >
          Fixed Price
        </option>
        <option
          value="Hourly-Rate"
        >
          Hourly Rate
        </option>
      </select>
      <h3>
        Estimated Budget
      </h3>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="$"
        required={true}
        type="number"
        value={454}
      />
      <h3>
        Phone Number
      </h3>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Phone"
        required={true}
        type="number"
        value="056565566"
      />
      <h3>
        Email
      </h3>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="email"
        required={true}
        type="email"
        value="TEST@ga.com"
      />
      <button
        className="button"
      >
        Post Project
      </button>
    </form>
  </div>
</div>
`;
